subsys/net/lib/lwm2m/lwm2m_obj_swmgmt.c

```diff
@@ -655,7 +655,7 @@ static struct lwm2m_engine_obj_inst *swmgmt_create(uint16_t obj_inst_id)
 	(void)memset(instance->package_name, 0, PACKAGE_NAME_LEN);
 	(void)memset(instance->package_version, 0, PACKAGE_VERSION_LEN);
 
-	instance->obj_inst_id = index;
+	instance->obj_inst_id = obj_inst_id;
```

Comment:
Good catch! In addition I think we also need to change `obj_inst_id` type to `uint16_t` in `struct lwm2m_swmgmt_data`.

Reply:
Fixed. Also renamed `i` and `j`